draft.markdown

```diff
@@ -0,0 +1,155 @@
+# Statistical Aggregations
+
+* Target Audience: Developers
+* Article Type: Development Frameworks
+
+## External References
+
+* https://www.elastic.co/guide/en/elasticsearch/reference/6.5/search-aggregations-metrics.html
+* https://www.elastic.co/guide/en/elasticsearch/reference/6.5/search-aggregations-metrics-cardinality-aggregation.html
+* https://lucene.apache.org/solr/guide/7_5/the-stats-component.html
+
+## Search Engine Connector Support
+* Elasticsearch 6: Yes
+* Solr 7: Yes
+
+## New/Related APIs
+
+Listed those which are relevant for developers.
+ 
+API (FQCN) | Provided by Artifact |
+---------: | :------------------: |
+`com.liferay.portal.search.searcher.SearchRequestBuilder#statsRequests(StatsRequest... statsRequests)` | com.liferay.portal.search.api
+`com.liferay.portal.search.searcher.SearchResponse#getStatsResponseMap()` | com.liferay.portal.search.api
+**`com.liferay.portal.search.stats.StatsRequest`** |	com.liferay.portal.search.api
+`com.liferay.portal.search.stats.StatsRequestBuilder` |	com.liferay.portal.search.api
+`com.liferay.portal.search.stats.StatsRequestBuilderFactory` |	com.liferay.portal.search.api
+**`com.liferay.portal.search.stats.StatsResponse`** |	com.liferay.portal.search.api
+`com.liferay.portal.kernel.search.Stats` | portal-kernel
+
+## Deprecated APIs
+* SearchSearchRequest#getStats()
+* SearchSearchRequest#setStats(Map<String, Stats> stats)
+
+----
+
+## Draft
+
+Support for [GroupBy](https://github.com/liferay/liferay-portal/blob/7.2.x/portal-kernel/src/com/liferay/portal/kernel/search/GroupBy.java) and [Stats](https://github.com/liferay/liferay-portal/blob/7.2.x/portal-kernel/src/com/liferay/portal/kernel/search/Stats.java) aggregations were introduced in 7.0.
+
+To extend Liferay's search aggregation capabilities, in @product-ver@, we added support for Cardinality Aggregations which provides an approximate (aka. statistical) count of distinct values. Please refer to the "External References" above for more details.

```

Comment:
Cardinality Aggregations sound like facet counts. Could they be used similarly?

For example, could a cardinality agggregation be used to compute the count of returned results matching the tags field value?

Reply:
Facets are "bucket aggregations", cardinality is different. The Elasticsearch cardinality stat guide covers the basics. I'd just redirect the users there.

Reply:
I was comparing a facet count to a cardinality aggregations. So a search query looks for 

> localhost:9200/liferay-*/_search?q=title_en_US=test

Couldn't you use a cardinality aggregation to count all the results with entryClassPK=33405 to count all the journal articles matching the query?

Reply:
No. That's two different things.

Faceted search is about executing a "main search" (keyword) then, collect terms from specific fields which represent the "facets" (aka. categories, filtering options etc.) so you can narrow down. And each bucket will have a certain number of docs = facet count.

Cardinality is about performing statistical stuff on top of the results.